inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.encode;
+
+import org.apache.inlong.sdk.transform.pojo.FieldInfo;
+import org.apache.inlong.sdk.transform.pojo.PbSinkInfo;
+import org.apache.inlong.sdk.transform.process.Context;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PbSinkEncoder implements SinkEncoder<byte[]> {
+
+    protected PbSinkInfo sinkInfo;
+
+    private Descriptors.Descriptor dynamicDescriptor;
+
+    private final Map<String, Descriptors.FieldDescriptor.Type> fieldTypes;
+
+    public PbSinkEncoder(PbSinkInfo pbSinkInfo) {
+        this.sinkInfo = pbSinkInfo;
+        this.fieldTypes = new HashMap<>();
+        for (FieldInfo field : pbSinkInfo.getFields()) {
+            fieldTypes.put(field.getName(), Descriptors.FieldDescriptor.Type.STRING);
+        }
+        // decode protoDescription
+        this.dynamicDescriptor = decodeProtoDescription(pbSinkInfo.getProtoDescription());
+    }
+
+    @Override
+    public byte[] encode(SinkData sinkData, Context context) {
+        try {
+            DynamicMessage.Builder dynamicBuilder = DynamicMessage.newBuilder(dynamicDescriptor);
```

Comment:
Output in ProtoBuf format, it is necessary to support the scenario where multiple pieces of data are merged together for encoding. There is no application scenario for encoding a single piece of data, right?

Reply:
I think of SinkData itself as a collection of data which has been transformed, and the output of transform is the multiple pieces of data that are merged together for encoding , if we need to encode them as pb we need to specify his description first, is my understanding correct?

Reply:
The other problem is how to encode the package header and summary fields.

Reply:
We can improve the sink type.